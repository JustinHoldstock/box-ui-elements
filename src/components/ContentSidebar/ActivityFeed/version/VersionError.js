/**
 * @flow
 * @file Version Error component
 */

import React, { ReactNode } from 'react';
import { FormattedMessage } from 'react-intl';

import messages from '../../../messages';

import './Version.scss';

function getErrorMessage(errorCode: string): ReactNode {
    switch (errorCode) {
        case 'tooManyVersions':
            return <FormattedMessage {...messages.versionTooManyVersions} />;
        default:
            return null;
    }
}

type Props = {
    errorCode: string
};

const VersionError = ({ errorCode }: Props): ReactNode => (
    <div className='bcs-version error'>
        <span className='bcs-version-message'>{getErrorMessage(errorCode)}</span>
    </div>
);

VersionError.displayName = 'VersionError';

export default VersionError;
