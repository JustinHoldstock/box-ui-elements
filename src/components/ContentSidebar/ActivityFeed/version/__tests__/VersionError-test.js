import React from 'react';
import { shallow } from 'enzyme';

import VersionError from '../VersionError';

describe('features/activity-feed/version/VersionError', () => {
    test('should correctly render error message', () => {
        const error = { errorCode: 'tooManyVersions' };

        const wrapper = shallow(<VersionError {...error} />);
        const formattedMessage = wrapper.find('FormattedMessage');

        expect(wrapper.hasClass('box-ui-version')).toBe(true);
        expect(wrapper.hasClass('error')).toBe(true);
        expect(formattedMessage.prop('defaultMessage')).toEqual('Multiple versions of this file');
    });
});
