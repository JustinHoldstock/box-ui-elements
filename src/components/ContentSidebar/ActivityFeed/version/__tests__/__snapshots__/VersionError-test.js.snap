// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/activity-feed/version/VersionError should correctly render error message 1`] = `
<div
  className="bcs-version error"
>
  <span
    className="bcs-version-message"
  >
    <FormattedMessage />
  </span>
</div>
`;
